protocol/tx/entry.go

```diff
@@ -26,10 +26,17 @@ type entry interface {
 var errInvalidValue = errors.New("invalid value")
 
 func entryID(e entry) (hash bc.Hash) {
+	// This nil test only handles the case where e is the zero value of
+	// the entry interface.
 	if e == nil {
 		return hash
 	}
 
+	// This nil test handles the case where e is a nil pointer.
+	if reflect.ValueOf(e).IsNil() {
```

Comment:
Is that idiomatic? I wonder whether it wouldn't be better to defend against the panic in `IsNil`:

```
if e == nil {
  return bc.Hash{}
}

v := reflect.ValueOf(e)
switch v.Kind() {
  case reflect.Chan, reflect.Func, reflect.Map, reflect.Ptr, reflect.Interface, reflect.Slice:
    if v.IsNil() {
      return bc.Hash{}
    }
}

...
```

(That could be factored out into a `SafeIsNil` or something.)